lib/ical/recur_iterator.js

```diff
@@ -262,77 +262,79 @@ class RecurIterator {
       this._nextByYearDay();
     }
 
-    if (this.rule.freq == "MONTHLY" && this.has_by_data("BYDAY")) {
-      let tempLast = null;
-      let initLast = this.last.clone();
-      let daysInMonth = Time.daysInMonth(this.last.month, this.last.year);
-
-      // Check every weekday in BYDAY with relative dow and pos.
-      for (let bydow of this.by_data.BYDAY) {
-        this.last = initLast.clone();
-        let [pos, dow] = this.ruleDayOfWeek(bydow);
-        let dayOfMonth = this.last.nthWeekDay(dow, pos);
-
-        // If |pos| >= 6, the byday is invalid for a monthly rule.
-        if (pos >= 6 || pos <= -6) {
-          throw new Error("Malformed values in BYDAY part");
-        }
+    if (this.rule.freq == "MONTHLY") {
+      if (this.has_by_data("BYDAY")) {
```

Comment:
Github's UI has failed to figure out that all I've done to this block is indent it. I'm pretty sure the block below (`else if (this.has_by_data("BYMONTHDAY")`) should also only be used if the rule is monthly.